### PR TITLE
Adds face concealing bandanas

### PR DESCRIPTION
**Pull request by Bartimeus973**
## About the PR 
Adds bandanas you can wear on your face to hide your identity (though its quite easy to guess who you actually are).
To make a bandana, use a handkerchief in hand. The bandana's color and motifs are dependant on the handkerchief. Using the bandana in hand will untie it to turn it into a handkerchief again.
Monkey and saurian friendly!

rootintootin

bandanasitems

bandanas

Wearing the bandana shows an ability that allows you to pull it up or down to wear it on your neck. Wearing it on your neck doesnt conceal your identity.
bandanadown

Yes it works on monkeys too!
monkeybandana

## Why's this needed? 
Flavorfull way of hiding your face. Completes a western look. Makes handkerchiefs more interesting and attractive.

```changelog
(u)Bartimeus973
(*)Adds face-covering bandanas to the game! Use a handkerchief in hand to do your best bandit impression!
```

### CODE REVIEW

**Comment by wolfolotl** [comment]
slight concern about making this kerchief-only since kerchiefs are spacebux items only (unless you're the ntsc) but aside from that I really like this

**Comment by Bartimeus973** [comment]
We could easily add the possibility to print handkerchiefs in general manufacturers.
Also due to cal's suggestion, i'll try to add a button to pull the bandana down and wear it on your neck.

**Comment by NightmareChamillian** [comment]
clothing manufacturers might be more in theme, but otherwise, I dig it.

File: code/obj/item/clothing/masks.dm [hunk 1]
```diff
@@ -626,3 +626,96 @@
...
+		if (ishuman(the_mob))
+			..()
+
+/obj/item/clothing/mask/bandana/abilities = list(/obj/ability_button/toggle_bandana)
+
+/obj/item/clothing/mask/bandana/attack_self(mob/user)
+	if (!src.handkerchief)
+		return
+	qdel(src)
+	var/obj/item/cloth/handkerchief/the_handkerchief = new src.handkerchief
```
Comment (pali6) [inline code/obj/item/clothing/masks.dm:648]:
```suggestion
	var/obj/item/cloth/handkerchief/the_handkerchief = new src.handkerchief
	the_handkerchief.color = src.color
```

File: code/obj/item/cloths.dm [hunk 2]
```diff
@@ -171,51 +171,79 @@ ABSTRACT_TYPE(/obj/item/cloth/handkerchief)
...
 /obj/item/cloth/handkerchief/attack(mob/living/M, mob/user)
 	if (!..())
 		return
 	user.visible_message("<span class='notice'>[user] [pick("dabs at", "blots at", "wipes")] [M == user ? his_or_her(user) : "[M]'s"] face with [src].</span>")
 
+/obj/item/cloth/handkerchief/attack_self(mob/user)
+	if (!src.bandana)
+		return
+	qdel(src)
+	var/obj/item/clothing/mask/bandana/the_bandana = new src.bandana
```
Comment (pali6) [inline code/obj/item/cloths.dm:185]:
```suggestion
	var/obj/item/clothing/mask/bandana/the_bandana = new src.bandana
	the_bandana = src.color
```

**Review by pali6 — APPROVED** [review]
This is cool :)
I'm adding color transferring because what we might do is to make white handkerchiefs craftable using clothing fabs and then people could paint them with paint.

**Pull request merged**